test/e2e/prometheusadapter_test.go

```diff
@@ -59,37 +59,11 @@ func isAPIServiceAvailable(conditions []apiservicesv1.APIServiceCondition) bool
 }
 
 func TestMetricsAPIAvailability(t *testing.T) {
```

Comment:
I'm not sure that we need this test anymore since TestPodMetricsPresence and TestNodeMetricsPresence already validate that the metrics API service is working.

Reply:
I think this test helps us to verify APIService

Reply:
If the API service isn't properly deployed then the metrics API will not work and the e2e tests exercising the API will fail. I feel that this particular test is redundant and I'd rather focus on user-facing scenarios.

Reply:
I see that in https://github.com/openshift/cluster-monitoring-operator/pull/2196, JP added a check into it to make sure it's pointing to the right service, that could be a good debugging test. But I agree that the check on the resource's status is redundant.

The test existed before `DedicatedServiceMonitors`, I think it's better to deal with it in another PR.